module_fcst_grid_comp.F90

```diff
@@ -455,157 +631,102 @@ subroutine fcst_initialize(fcst_comp, importState, exportState, clock, rc)
-        glonPtr(tlb(1):tub(1),tlb(2):tub(2)) = &
-           Atmos%lon_bnd(tlb(1):tub(1),tlb(2):tub(2)) * dtor
-        call ESMF_GridGetCoord(fcstGrid, coordDim=2, staggerLoc=ESMF_STAGGERLOC_CORNER, &
-                               totalLBound=tlb, totalUBound=tub, &
-                               farrayPtr=glatPtr, rc=rc); ESMF_ERR_ABORT(rc)
-        glatPtr(tlb(1):tub(1),tlb(2):tub(2)) = &
-          Atmos%lat_bnd(tlb(1):tub(1),tlb(2):tub(2)) * dtor
-
-        call mpp_error(NOTE, 'after create fcst grid for regional-only')
-
-      else ! not regional only
-
-        if (.not. Atmos%regional .and. .not. Atmos%nested ) then  !! global only
-
-          do tl=1,6
-              decomptile(1,tl) = Atmos%layout(1)
-              decomptile(2,tl) = Atmos%layout(2)
-              decompflagPTile(:,tl) = (/ESMF_DECOMP_SYMMEDGEMAX,ESMF_DECOMP_SYMMEDGEMAX/)
-          enddo
-          fcstGrid = ESMF_GridCreateMosaic(filename="INPUT/"//trim(gridfile),                                   &
-                                             regDecompPTile=decomptile,tileFilePath="INPUT/",                   &
-                                             decompflagPTile=decompflagPTile,                                   &
-                                             staggerlocList=(/ESMF_STAGGERLOC_CENTER, ESMF_STAGGERLOC_CORNER/), &
-                                             name='fcst_grid', rc=rc)
+          fcstGridComp(n) = ESMF_GridCompCreate(name="global", petList=pelist, rc=rc); ESMF_ERR_ABORT(rc)
+
+          call ESMF_InfoGetFromHost(fcstGridComp(n), info=info, rc=rc); ESMF_ERR_ABORT(rc)
+          call ESMF_InfoSet(info, key="layout", values=layout, rc=rc); ESMF_ERR_ABORT(rc)
+          call ESMF_InfoSet(info, key="tilesize", value=Atmos%mlon, rc=rc); ESMF_ERR_ABORT(rc)
+
+          call ESMF_GridCompSetServices(fcstGridComp(n), SetServicesNest, userrc=urc, rc=rc)
           if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+          if (ESMF_LogFoundError(rcToCheck=urc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__, rcToReturn=rc)) return
 
-          call mpp_error(NOTE, 'after create fcst grid for global-only with INPUT/'//trim(gridfile))
+        else
 
-        else !! global-nesting or regional-nesting
+          allocate(petListNest(layout(1)*layout(2)))
+          k=pelist(1)
+          do j=1,layout(2)
+          do i=1,layout(1)
+             petListNest(k-pelist(1)+1) = k
+             k = k + 1
+          end do
+          end do
 
-          if (mype == 0) TileLayout = Atmos%layout
-          call ESMF_VMBroadcast(vm, bcstData=TileLayout, count=2, &
-                                  rootPet=0, rc=rc)
-          if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+          fcstGridComp(n) = ESMF_GridCompCreate(name="nest", petList=petListNest, rc=rc); ESMF_ERR_ABORT(rc)
```

Comment:
@danrosen25 Please see Gerhard's comment above, if you rename the SetServicesNest, maybe rename the name here too as it contains regional top domain too.